clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -199,9 +199,19 @@ public void logUnused() {
         return t.cast(o);
     }
 
+    /**
+     * Get a list of configured instances of the given class specified by the given configuration key. The configuration
+     * may specify either null or an empty string to indicate no configured instances. In both cases, this method
+     * returns an empty list to indicate no configured instances.
+     * @param key The configuration key for the class
+     * @param t The interface the class should implement
+     * @return The list of configured instances
+     */
     public <T> List<T> getConfiguredInstances(String key, Class<T> t) {
         List<String> klasses = getList(key);
         List<T> objects = new ArrayList<T>();
+        if (klasses == null)
+            return objects;
```

Comment:
so currently all uses of this method expect empty list (since null was never returned), So I did that not to break current uses. I can remove that and return null, but than I have to fix all other uses -- I thought that was too big of a change.